### ISSUE TITLE
Pricing pane

### ISSUE THREAD

**Issue by blitzmann** [pyfa-org/Pyfa#1158]
https://forums.eveonline.com/default.aspx?g=posts&m=6937833#post6937833

A user on the eve-o forums brought up a good point about the new pricing pane, and one that I was concerned about as well before we added it. Implants and boosters are special in that they aren't really a fitting thing, and thus for many users they don't care.

The idea that I have is to default to minimal (original pane), and have a tooltip that breaks down the other things. Full version can remain as is I believe. Need to look into this a bit more, and gather some feedback from others.

**Comment by petosorus** [comment]
Hi!
I'm interested in this (seems I only do money things in pyfa...)

How about not including the "character" price in the total but still leaving it displayed?
Could be the same with cargo and drones. In my personal use, I have a stack of ammo and a stack of drones that I just use to fill the ship, a bit like what this user says about boosters. That's personal of course, and I would like feedback.

I'm thinking about something like this 
Ship    |    Fittings    |    Total (Ship + fittings)
Cargo |    Drones    |    Character

Another solution could be to keep only the original pane, and have a 'consumables' total with the Cargo|Drones|Char breakdown as a tooltip

**Comment by blitzmann** [comment]
@petosorus 

> I'm thinking about something like this
>Ship | Fittings | Total (Ship + fittings)
>Cargo | Drones | Character

The only thing about this is that I believe Drones should be included. A lot of time (at least when I played EVE >_>), if I went out, I fully expected to lose everything, which includes drones. So total, for me, would be a "fuck, I died... what do I have to pay to get back to where I was". But then the same can be said about cargo as well, but then sometimes cargo isn't really a that should be includes and sometimes it is

As a quick fix, I would simply exclude character from the total and note that in the tooltip for total...

However, with a bit more effort, I think we can make the total be pretty customization. I thought about this a bit more a couple nights ago, and I was thinking having a context menu that the user can then select which entities they want included in the total. So, it would be like 

```
____________________________
| ✓ | Hull and fittings    |
| ✓ | Drones               |
|    |  Cargo              |
|    | Implants            |
|    | Boosters            |
___________________________
```

This can be done with a wx menu pretty easily, and the values can be saved in the user's preferences via a bitmask (so each option would be a specific bit in the bitfield, which would toggle on/off).

Thoughts?

**Comment by Ebag333** [comment]
Talking with a few people on Slack, they were all pretty okay with minimal being just "above the line" and full being everything.

Making it optional via menu selections would work fine, but I think that the majority of folks will be okay with hull + modules, especially as that's what the icons match.  It was always slightly weird to include some things as "modules" and some things as not.

>The only thing about this is that I believe Drones should be included. A lot of time (at least when I played EVE >_>), if I went out, I fully expected to lose everything, which includes drones. So total, for me, would be a "fuck, I died... what do I have to pay to get back to where I was". But then the same can be said about cargo as well, but then sometimes cargo isn't really a thing that should be includes and sometimes it is.

Like you said, it's all highly variable from person to person _and_ from fit to fit.  So I'd either differentiate between above the line (no tabs) and full, or full customizable.

The down side to fully customizable is that it's a lot more complicated, and it'd only apply to the minimal pane (no reason to apply it to the full), so it's certainly possible to cause confusion for users.

**Comment by blitzmann** [comment]
All good points, and all things I can get behind as well.

> The down side to fully customizable is that it's a lot more complicated, and it'd only apply to the minimal pane (no reason to apply it to the full), so it's certainly possible to cause confusion for users.

I wouldn't say it only applies to minimal pane. For example, I may want to see the booster / implant cost at a glance, but personally don't care about that being included in total. It is slightly more complicated to build out, and I think there should be sane defaults for the total which most folks would probably keep. But at the same time it does allow folks to pick and choose what they want to see included in the total at any given time / session / mood (instead of having to switch to min/full, then restart pyfa). It's all very personal and subjective, and it's not possible to make everyone happy with any hardcoded decisions, which is why I think a menu for total might be a good fit. Then you have tooltips, which I think should be a good breakdown on everything anyway.

I think I would personally prefer either the menu option or, as a quick and simple fix, simply keep the total the same for both full and minimal (ship + modules), and display the "extras" for full view, along with tooltips giving a cost breakdown. Burt I'm down for whatever @petosorus decides he wants to tackle. Lots of options on how to proceed here; all right, none wrong, which is better than the other way around :D

**Comment by petosorus** [comment]
I like the idea of the context menu to set options of what's in total and what's not.
It would be accessible on both full and minimal panes, and be taken into account for both totals

**Comment by blitzmann** [comment]
This issue was addressed with @petosorus contribution (#1194) and will be available in the 1.30.x release cycle. A generated build which includes this addition can be found here: https://ci.appveyor.com/project/blitzmann/pyfa/build/development-118/artifacts

Thanks @petosorus!
